modules/virtual_network_gateway/variables.tf

```
@@ -136,59 +129,95 @@ variable "ip_configuration" {
       subnet_id        = "ID_for_subnet_GatewaySubnet"
     }
   ]
-
+  ```
   EOF
-  type        = list(any)
+  default     = []
+  nullable    = false
+  type = list(object({
+    name                          = optional(string, "vnetGatewayConfig")
+    create_public_ip              = bool
+    public_ip_name                = optional(string)
+    private_ip_address_allocation = optional(string, "Dynamic")
+    public_ip_standard_sku        = optional(bool, false)
+    subnet_id                     = string
+  }))
 }
 
 variable "vpn_client_configuration" {
   description = <<-EOF
-  List of VPN client configurations - every object in the list contains attributes:
-  - address_space - the address space out of which IP addresses for vpn clients will be taken. You can provide more than one address space, e.g. in CIDR notation.
-  - aad_tenant - AzureAD Tenant URL
-  - aad_audience - the client id of the Azure VPN application. See Create an Active Directory (AD) tenant for P2S OpenVPN protocol connections for values
-  - aad_issuer - the STS url for your tenant
-  - root_certificate - one or more root_certificate blocks which are defined below. These root certificates are used to sign the client certificate used by the VPN clients to connect to the gateway.
-  - revoked_certificate - one or more revoked_certificate blocks which are defined below.
-  - radius_server_address - the address of the Radius server.
-  - radius_server_secret - the secret used by the Radius server.
-  - vpn_client_protocols - list of the protocols supported by the vpn client. The supported values are SSTP, IkeV2 and OpenVPN. Values SSTP and IkeV2 are incompatible with the use of aad_tenant, aad_audience and aad_issuer.
-  - vpn_auth_types - list of the vpn authentication types for the virtual network gateway. The supported values are AAD, Radius and Certificate.
+  VPN client configurations (IPSec point-to-site connections).
+
+  List of available attributes of each VPN client configurations:
+  - `address_space`           - (`string`, required) the address space out of which IP addresses for vpn clients will be taken. You can provide more than one address space, e.g. in CIDR notation.
+  - `aad_tenant`              - (`string`, optional) AzureAD Tenant URL
```

I would suggest the following whenever we have optional value, even if it's `null`
```suggestion
  - `aad_tenant`              - (`string`, optional, defaults to `null`) AzureAD Tenant URL
```

---

ok, fixed this and the others optionals too